### PR TITLE
LPS-136716 Provide balloon editor configuration in JSON files

### PR DESCRIPTION
**Pull request by diegonvs**
From: https://issues.liferay.com/browse/LPS-136716

# Context: 

Currently, the default configuration is duplicated, in:

* CKEditorBalloonEditorConfigContributor.java . This is applied if BE is used through `<liferay-editor:editor>` tag. It overrides JS configuration.
* BalloonEditor.js . This is applied if BE is used directly, for example in a React component. In that case, `config` prop will still have priority.
 

Something we might want to consider, as a part of this task, is to remove all configuration from `CKEditorBalloonEditorConfigContributor.java`, just letting it inherit configuration in JS/JSON file.

# How to test it?

1.  Deploy `frontend-js-ckeditor-web` with this change
2.  Deploy `frontend-editor-ckeditor-sample-web`
3. See balloon editor with all panels working :)

### CODE REVIEW

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/editor/configuration/CKEditorBalloonEditorConfigContributor.java [hunk 1]
```diff
@@ -56,87 +42,6 @@ public void populateConfigJSONObject(
 		super.populateConfigJSONObject(
```
Comment (diegonvs) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/java/com/liferay/frontend/editor/ckeditor/web/internal/editor/configuration/CKEditorBalloonEditorConfigContributor.java:42]: We still needing CKEditorBalloonEditorConfigContributor class to provide the dynamic values like: links to DL, portal language, direction, resolved CSS info, [etc.](https://liferay.atlassian.net/wiki/spaces/~73202423/pages/1810106835/Provide+editor+configuration+in+JSON+files#Required-Server-Side-Properties)

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js [hunk 2]
```diff
@@ -16,30 +16,20 @@ import PropTypes from 'prop-types';
...
-			'ImageAlignLeft,ImageAlignCenter,ImageAlignRight,LinkAddOrEdit,AltImg',
-		toolbarTable: 'TableHeaders,TableRow,TableColumn,TableCell,TableDelete',
-		toolbarText:
-			'Styles,Bold,Italic,Underline,BulletedList,NumberedList,TextLink,' +
-			'JustifyLeft,JustifyCenter,JustifyRight,LineHeight,RemoveFormat',
-		toolbarVideo: 'VideoAlignLeft,VideoAlignCenter,VideoAlignRight',
-	};
-
+const BalloonEditor = ({
+	baseConfig = DEFAULT_BALLOON_EDITOR_CONFIG,
```
Comment (markocikos) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js:24]: Does `baseConfig` need to be a prop? :thinking: `config` should be enough to allow setting config overrides.

**Review by markocikos — COMMENTED** [review]
@diegonvs See one question inline.

In addition:
- A minor gripe: commit message is `Provide balloon editor configuration in JSON files`, but we are not providing it in JSON files :slightly_smiling_face: 
- This PR will be in conflict with https://github.com/liferay-frontend/liferay-portal/pull/1350. Please hold forwarding one if the other is forwarding.

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js (hunk 2 above)
Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js:24]: image

Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js:24]: image

Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js:24]: > Does baseConfig need to be a prop?

Nope! Removing!

Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js:24]: force-pushing!

**Comment by diegonvs** [comment]
About `baseConfig` and commit_msg issue, see: https://github.com/liferay-frontend/liferay-portal/pull/1352/commits/19d98168a13fda64e4e29e25c43ebe3bdd685c89 :)

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js (hunk 2 above)
Comment (diegonvs) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/editor/BalloonEditor.js]: https://github.com/liferay-frontend/liferay-portal/pull/1352/commits/19d98168a13fda64e4e29e25c43ebe3bdd685c89

**Comment by diegonvs** [comment]
ci:test:sf

**Comment by diegonvs** [comment]
ci:test:relevant

**Comment by markocikos** [comment]
ci:forward

**Review by markocikos — APPROVED** [review]

**Pull request closed without merging**